pkg/tcpip/network/fragmentation/reassembler.go

```diff
@@ -94,6 +97,18 @@ func (r *reassembler) process(first, last uint16, more bool, vv buffer.Vectorise
 		consumed = vv.Size()
 		r.size += consumed
 	}
+
+	// Invoking a goroutine to start a timer and to notify after 30 seconds
+	// for checking whether any of fragment is missing.
+	// If fragment is missing, Invoke a TimeOut handler.
+	go func() {
+		<-timer.C
+		if r.deleted < len(r.holes) {
+			r.TimeOut(stack, netHeader, vv)
+			return
+		}
```

Comment:
I believe we don't need a new goroutine. We can just use an AfterFunc() to run the timer after 30 seconds. If we finish reassembling before that then we can just cancel the timer.

Reply:
Also this should set the reassembler in an error state so that a future call to Process fails and causes the reassembler to be cleaned up?

Reply:
we have taken care both of above comments.